csharp/src/Microsoft.ML.OnnxRuntime/InferenceSession.cs

```diff
@@ -993,9 +998,171 @@ public bool IsTensor
+                }
+
+
+                // Process domain
+
+                NativeApiStatus.VerifySuccess(NativeMethods.OrtModelMetadataGetDomain(modelMetadataHandle, allocator.Pointer, out domainHandle));
+                using (var ortAllocation = new OrtMemoryAllocation(allocator, domainHandle, 0))
+                {
+                    _domain = NativeOnnxValueHelper.StringFromNativeUtf8(domainHandle);
+                }
+
+                // Process description
+
+                NativeApiStatus.VerifySuccess(NativeMethods.OrtModelMetadataGetDescription(modelMetadataHandle, allocator.Pointer, out descriptionHandle));
+                using (var ortAllocation = new OrtMemoryAllocation(allocator, descriptionHandle, 0))
+                {
+                    _description = NativeOnnxValueHelper.StringFromNativeUtf8(descriptionHandle);
+                }
+
+                // Process version
+
+                NativeApiStatus.VerifySuccess(NativeMethods.OrtModelMetadataGetVersion(modelMetadataHandle, out _version));
+
+
+                // Process CustomMetadata Map
+
+                IntPtr customMetadataMapKeysHandle = IntPtr.Zero;
+                long numKeys;
+                NativeApiStatus.VerifySuccess(NativeMethods.OrtModelMetadataGetCustomMetadataMapKeys(modelMetadataHandle, allocator.Pointer, out customMetadataMapKeysHandle, out numKeys));
+
+                // We have received an array of null terminated C strings which are the keys that we can use to lookup the custom metadata map
+                // The OrtAllocator will finally free the customMetadataMapKeysHandle
+                using (var ortAllocation = new OrtMemoryAllocation(allocator, customMetadataMapKeysHandle, 0))
+                {
+                    for (int i = 0; i < (int)numKeys; ++i)
+                    {
+                        IntPtr keyHandle = IntPtr.Zero;
+                        IntPtr valueHandle = IntPtr.Zero;
+
+                        try
+                        {
+                            keyHandle = Marshal.ReadIntPtr(customMetadataMapKeysHandle, IntPtr.Size * i);
+
+                            NativeApiStatus.VerifySuccess(NativeMethods.OrtModelMetadataLookupCustomMetadataMap(modelMetadataHandle, allocator.Pointer, keyHandle, out valueHandle));
+
+                            var key = NativeOnnxValueHelper.StringFromNativeUtf8(keyHandle);
+                            var value = NativeOnnxValueHelper.StringFromNativeUtf8(valueHandle);
+
+                            // Put the key/value pair into the dictionary
+                            _customMetadataMap[key] = value;
+
+                        }
+
+                        finally
+                        {
+                            // Free the memory associated with this single key value pair
+                            allocator.FreeMemory(keyHandle);
+                            allocator.FreeMemory(valueHandle);
```

Comment:
Why not wrap it in using() with OrtMemoryAllocation()? Simlper code

Reply:
Yes, have that now - thank you!